kolibri/plugins/management/assets/src/vuex/actions.js

```diff
@@ -10,93 +10,62 @@ function setSelectedGroupId({ dispatch }, id) {
   dispatch('SET_SELECTED_GROUP_ID', id);
 }
 
-/*
-Used to wrap xhrs below, but note that it sets the onreadystatechange function,
-trampling whatever is there.
- */
-function promiseWrapper(xhr) {
-  return new Promise((resolve, reject) => {
-    xhr.onreadystatechange = () => { // eslint-disable-line no-param-reassign
-      if (xhr.readyState === 4 && xhr.status === 200) {
-        resolve();
-      } else if (xhr.readyState === 4 && xhr.status !== 200) {
-        reject();
-      }
-    };
-  });
-}
+const Resource = require('../../../../../core/assets/src/api_resource');
 
-// An action for setting up the initial state of the app by fetching data from the server
-function fetch({ dispatch }, classroomListUrl, learnerGroupListUrl,
-                             learnerListUrl, membershipListUrl) {
-  /*
-  Takes as parameters urls for the four API list endpoints that it uses.
-   */
-  const xhr1 = new XMLHttpRequest();
-  const xhr2 = new XMLHttpRequest();
-  const xhr3 = new XMLHttpRequest();
-  const xhr4 = new XMLHttpRequest();
+const classroomCollection = new Resource.Collection([], 'classroom');
+const learnerGroupCollection = new Resource.Collection([], 'learnergroup');
+const learnerCollection = new Resource.Collection([], 'facilityuser');
+const memberCollection = new Resource.Collection([], 'membership');
 
-  const promises = [xhr2, xhr3, xhr4].map(xhr => promiseWrapper(xhr));
-
-  xhr1.open('GET', classroomListUrl);
-  let classrooms = [];
-  xhr1.onreadystatechange = () => {
-    if (xhr1.readyState === 4 && xhr1.status === 200) {
-      classrooms = JSON.parse(xhr1.response);
-      const cids = classrooms.map(c => c.id);
-      xhr2.open(
-        'GET',
-        `${learnerGroupListUrl}?parent_in=${global.encodeURIComponent(JSON.stringify(cids))}`
-      );
-      xhr2.send();
-    }
-  };
-  xhr1.send();
+// An action for setting up the initial state of the app by fetching data from the server
```

Comment:
I know this is just an adaptation of existing code, but a lot of the data fetching/filtering idioms in this file are a bit strange and possibly inefficient. Overall, it also feels very boiler-platey, and it seems like we could probably find a higher level of abstraction for use in writing actions. In particular, I'm thinking about related models and how we handle them. Let's plan a followup brainstorming session around this, perhaps?